homeassistant/components/thethingsnetwork/sensor.py

```diff
@@ -78,7 +78,7 @@ def state(self):
         if self._ttn_data_storage.data is not None:
             try:
                 return round(self._state[self._value], 1)
-            except KeyError:
+            except (KeyError, TypeError):
                 pass
```

Comment:
We should `return None` here.

Reply:
Also `return None` in the last return case, `if self._ttn_data_storage.data is None`.

Reply:
Yeah, the state should be `None`.